src/Raven.Server/Documents/ResourceCache.cs

```diff
@@ -72,8 +72,23 @@ private bool UnlikelyTryGet(StringSegment resourceName, out Task<TResource> reso
             if (_caseInsensitive.TryGetValue(resourceName, out resourceTask) == false)
                 return false;
 
+            _forTestingPurposes?.OnUnlikelyTryGet?.Invoke();
+
             lock (this)
             {
+                if (_caseInsensitive.TryGetValue(resourceName, out var resourceTaskUnderLock) == false)
+                {
+                    // database was deleted
+                    return false;
+                }
+
+                if (resourceTask != resourceTaskUnderLock)
+                {
+                    // we have a case in sensitive match, but it is not the same instance
+                    resourceTask = resourceTaskUnderLock;
+                    return true;
```

Comment:
Can we have a test case which results in hitting this condition?

Reply:
done